Refactor: improve introductory texts (#112)
decentralized-identity/.well-known: README.md

### Please DO NOT contact IANA regarding registration of well known URIs associated with these specifications.

# Well-Known DID Configuration

> This specification defines a way for a website to prove that its DID is verifiably linked to that website. Well-Known DID configuration builds on and is an extension to the core [DID specification](https://www.w3.org/TR/did-1.0/). Furthermore, it  establishes the `/.well-known/did-configuration.json` URI pattern, which is [IANA registered](https://www.iana.org/assignments/well-known-uris/well-known-uris.xhtml).

Reviewer: ```suggestion
This specification defines a way for a website to prove that its DID is verifiably linked to that website. Well-Known DID configuration builds on and is an extension to the core [DID specification](https://www.w3.org/TR/did-1.0/). Furthermore, it establishes the [IANA registered](https://www.iana.org/assignments/well-known-uris/well-known-uris.xhtml) `/.well-known/did-configuration.json` URI pattern.
```

> > Note: The examples in this spec mostly make use of the [did:web Method specification](https://w3c-ccg.github.io/did-method-web/). However, this specification works with any DID method supporting the `services` property in the [DID document](https://www.w3.org/TR/did-1.0/#services) (e.g. did:jwk and [did:key](https://w3c-ccg.github.io/did-key-spec/) don't support this, [did:web](https://w3c-ccg.github.io/did-method-web/) and [did:iota](https://docs.iota.org/developer/iota-identity/references/iota-did-method-spec) do support it)

Reviewer: ```suggestion
> Note: The examples in this spec mostly make use of the [`did:web` Method specification](https://w3c-ccg.github.io/did-method-web/). However, this specification is compatible with any DID method supporting the [`services` property](https://www.w3.org/TR/did-1.0/#services) in the DID document (e.g., `did:jwk` and [`did:key`](https://w3c-ccg.github.io/did-key-spec/) don't support this; [`did:web`](https://w3c-ccg.github.io/did-method-web/) and [`did:iota`](https://docs.iota.org/developer/iota-identity/references/iota-did-method-spec) do support it).
```

## What this solves

> The problem that was unsolved in the [DID core specification]() as well as in the [did:web Method specification](https://w3c-ccg.github.io/did-method-web/) is verifiably proving that a DID belongs to a website and vice versa. To be more specific this specification links an [origin](https://datatracker.ietf.org/doc/html/rfc6454) with a DID. This means a DID cannot be linked to any website URL with path segments for example, only to the website [origin](https://datatracker.ietf.org/doc/html/rfc6454).

Reviewer: ```suggestion
One problem that was left unsolved by both the [DID core specification]() and the [`did:web` Method specification](https://w3c-ccg.github.io/did-method-web/) is how to verifiably prove that a DID belongs to a website and vice versa. To be more specific, this specification supports linking an [origin](https://datatracker.ietf.org/doc/html/rfc6454) to a DID. This means, for example, that this specification does not support linking a DID to any website URL with path segments, only to the website [origin](https://datatracker.ietf.org/doc/html/rfc6454).
```

> This specification makes this possible by building on the `LinkedDomains` service of the [DID document](https://www.w3.org/TR/did-1.0/#services). The `serviceEndpoint` field contains the website [origin](https://datatracker.ietf.org/doc/html/rfc6454), which in turn hosts the `/.well-known/did-configuration.json`. The `did-configuration.json` file then holds a [Verifiable Credential](https://www.w3.org/TR/vc-data-model/) (VC) claiming the [origin](https://datatracker.ietf.org/doc/html/rfc6454). This [Verifiable Credential](https://www.w3.org/TR/vc-data-model/) must be signed with key material from the original [DID document](https://www.w3.org/TR/did-1.0/#services) which contains the `LinkedDomains` service. 

Reviewer: ```suggestion
This specification supports such linkage by building on the `LinkedDomains` service of the [DID document](https://www.w3.org/TR/did-1.0/#services). The `serviceEndpoint` field contains the website [origin](https://datatracker.ietf.org/doc/html/rfc6454), which in turn hosts the `/.well-known/did-configuration.json`. The `did-configuration.json` file then holds a [Verifiable Credential](https://www.w3.org/TR/vc-data-model/) (VC) claiming the [origin](https://datatracker.ietf.org/doc/html/rfc6454). This [Verifiable Credential](https://www.w3.org/TR/vc-data-model/) must be signed with key material from the original [DID document](https://www.w3.org/TR/did-1.0/#services) which contains the `LinkedDomains` service. 
```

> > Note: The DID document can be hosted at the [IANA registered](https://www.iana.org/assignments/well-known-uris/well-known-uris.xhtml) well-known endpoint of the same origin, `/.well-known/did.json`, but this is specific to the [did:web Method specification](https://w3c-ccg.github.io/did-method-web/). The DID configuration however, works with any did method supporting the DID document services and these DID documents can be hosted anywhere, not necessarily at `/.well-known/did.json`.

Reviewer: ```suggestion
> Note: The DID document can be hosted at the [IANA registered](https://www.iana.org/assignments/well-known-uris/well-known-uris.xhtml) well-known endpoint of the same origin, `/.well-known/did.json`, but this is specific to the [`did:web` Method specification](https://w3c-ccg.github.io/did-method-web/). The DID configuration, however, works with any DID method supporting the DID document services, and these DID documents can be hosted anywhere, not only at `/.well-known/did.json`.
```

> ## .well-known/did-configuration.json

Reviewer: ```suggestion
## `.well-known/did-configuration.json`
```

### Known implementations

> | implementer  | did.json                                                   | did-configuration.json                                                   | date last checked |

Reviewer: ```suggestion
| implementer  | `did.json`                                                 | `did-configuration.json`                                                 | date last checked |
| ------------ | ---------------------------------------------------------- | ------------------------------------------------------------------------ | ----------------- |
```

> | ------------ | ---------------------------------------------------------- | ------------------------------------------------------------------------ | ----------------- |

Reviewer: ```suggestion
| ------------ |:----------------------------------------------------------:|:------------------------------------------------------------------------:|:-----------------:|
```

> | dif          | [X](https://identity.foundation/.well-known/did.json)      | [X](https://identity.foundation/.well-known/did-configuration.json)      | 12-10-20          |
> | transmute    | [X](https://www.transmute.industries/.well-known/did.json) | [X](https://www.transmute.industries/.well-known/did-configuration.json) | 12-10-20          |
> | mattr global | [X](https://mattr.global/.well-known/did.json)             |                                                                          | 12-10-20          |
> | trinsic      | [X](https://trinsic.id/.well-known/did.json)               | [X](https://trinsic.id/.well-known/did-configuration.json)               | 12-15-20          |

Reviewer: ```suggestion
| dif          | [X](https://identity.foundation/.well-known/did.json)      | [X](https://identity.foundation/.well-known/did-configuration.json)      | 2020-12-10        |
| transmute    | [X](https://www.transmute.industries/.well-known/did.json) | [X](https://www.transmute.industries/.well-known/did-configuration.json) | 2020-12-10        |
| mattr global | [X](https://mattr.global/.well-known/did.json)             |                                                                          | 2020-12-10        |
| trinsic      | [X](https://trinsic.id/.well-known/did.json)               | [X](https://trinsic.id/.well-known/did-configuration.json)               | 2020-12-15        |
```
